raphtory/src/algorithms/embeddings/fast_rp.rs

```diff
@@ -0,0 +1,427 @@
+                    0.0,
+                    1.6817928305074292,
+                    -1.6817928305074292,
+                    -1.6817928305074292,
+                    -1.6817928305074292,
+                    0.0,
+                    0.0,
+                    0.0,
+                    -1.6817928305074292,
+                    1.6817928305074292,
+                    0.0,
+                    0.0,
+                    0.0,
+                ],
+            ),
+            (
+                "3",
+                [
+                    1.6817928305074292,
+                    1.6817928305074292,
+                    -1.6817928305074292,
+                    0.0,
+                    0.0,
+                    3.3635856610148585,
+                    1.6817928305074292,
+                    1.6817928305074292,
+                    3.3635856610148585,
+                    -3.3635856610148585,
+                    0.0,
+                    1.6817928305074292,
+                    0.0,
+                    -1.6817928305074292,
+                    0.0,
+                    -3.3635856610148585,
+                ],
+            ),
+            (
+                "1",
+                [
+                    1.6817928305074292,
+                    1.6817928305074292,
+                    -1.6817928305074292,
+                    0.0,
+                    0.0,
+                    3.3635856610148585,
+                    1.6817928305074292,
+                    1.6817928305074292,
+                    3.3635856610148585,
+                    -3.3635856610148585,
+                    0.0,
+                    1.6817928305074292,
+                    0.0,
+                    -1.6817928305074292,
+                    0.0,
+                    -3.3635856610148585,
+                ],
+            ),
+            (
+                "4",
+                [
+                    0.0,
+                    3.3635856610148585,
+                    -1.6817928305074292,
+                    -1.6817928305074292,
+                    1.6817928305074292,
+                    -1.6817928305074292,
+                    1.6817928305074292,
+                    0.0,
+                    -1.6817928305074292,
+                    0.0,
+                    0.0,
+                    -1.6817928305074292,
+                    1.6817928305074292,
+                    1.6817928305074292,
+                    -1.6817928305074292,
+                    -1.6817928305074292,
+                ],
+            ),
+        ]);
+
+        test_storage!(&graph, |graph| {
+            let results =
+                fast_rp(graph, 16, 1.0, vec![1.0, 1.0], Some(42), None).get_all_with_names();
+            for (v_id, embedding) in results {
+                assert_eq!(embedding, *baseline.get(v_id.as_str()).unwrap());
+            }
+        });
+    }
+
+    // NOTE(Wyatt): the simple fast_rp test is more of a validation of idempotency than correctness (although the results are expected)
```

Comment:
Same as this here - wanna create a fastRP 2.0 ticket to track this

Reply:
On the point of extra testing, perhaps some window tests + layer tests etc - this is probably a good idea for us to add within the larger test pipeline

Reply:
Also yes.